janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/strategy/AdjacentVertexFilterOptimizerStrategy.java

```diff
@@ -111,12 +111,13 @@ private OptimizableQueryType analyzeSubSteps(List<Step> steps) {
         } else if (steps.get(1) instanceof HasStep) {
             // Check if this filter traversal matches the pattern: _.inV/outV/otherV.hasId(x)
             HasStep hasStep = (HasStep) steps.get(1);
-            if (hasStep.getHasContainers().size() != 1) {
+            List hasContainers = hasStep.getHasContainers();
```

Comment:
nitpick:
```suggestion
            List<HasContainer> hasContainers = hasStep.getHasContainers();
```
so that you don't need to do casting on line 120, i.e. `(HasContainer) hasContainers.get(0);`

Reply:
Thank you for suggestion. Resolved